File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        int x = root.val;
        if (p.val > x && q.val > x){
            return lowestCommonAncestor(root.right, p, q);
        }
         if (p.val < x && q.val < x){
            return lowestCommonAncestor(root.left, p, q);
        }
        return root;
    }
}

File: Data Structures & Algorithms/path-sum/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root is None:
            return False
        left_sum = targetSum - root.val
        if root.left is None and root.right is None and left_sum == 0:
            return True
        return self.hasPathSum(root.left, left_sum) or self.hasPathSum(root.right, left_sum)
